js/pro/deribit.js

```diff
@@ -0,0 +1,843 @@
+         * @param {object} params extra parameters specific to the deribit api endpoint
+         * @param {str|undefined} params.interval specify aggregation and frequency of notifications. Possible values: 100ms, raw
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const url = this.urls['api']['ws'];
+        const interval = this.safeString (params, 'interval', '100ms');
+        params = this.omit (params, 'interval');
+        const channel = 'trades.' + market['id'] + '.' + interval;
+        if (interval === 'raw') {
+            await this.authenticate ();
+        }
+        const message = {
+            'jsonrpc': '2.0',
+            'method': 'public/subscribe',
+            'params': {
+                'channels': [ channel ],
+            },
+            'id': this.requestId (),
+        };
+        const request = this.deepExtend (message, params);
+        const trades = await this.watch (url, channel, request, channel, request);
+        return this.filterBySinceLimit (trades, since, limit, 'timestamp', true);
+    }
+
+    handleTrades (client, message) {
+        //
+        //     {
+        //         "jsonrpc": "2.0",
+        //         "method": "subscription",
+        //         "params": {
+        //             "channel": "trades.BTC_USDC-PERPETUAL.100ms",
+        //             "data": [{
+        //                 "trade_seq": 501899,
+        //                 "trade_id": "USDC-2436803",
+        //                 "timestamp": 1655397355998,
+        //                 "tick_direction": 2,
+        //                 "price": 21026,
+        //                 "mark_price": 21019.9719,
+        //                 "instrument_name": "BTC_USDC-PERPETUAL",
+        //                 "index_price": 21031.7847,
+        //                 "direction": "buy",
+        //                 "amount": 0.049
+        //             }]
+        //         }
+        //     }
+        //
+        const params = this.safeValue (message, 'params', {});
+        const channel = this.safeString (params, 'channel', '');
+        const parts = channel.split ('.');
+        const marketId = this.safeString (parts, 1);
+        const symbol = this.safeSymbol (marketId);
+        const market = this.safeMarket (marketId);
+        const trades = this.safeValue (params, 'data', []);
+        let stored = this.safeValue (this.trades, symbol);
+        if (stored === undefined) {
+            const limit = this.safeInteger (this.options, 'tradesLimit', 1000);
+            stored = new ArrayCache (limit);
+            this.trades[symbol] = stored;
+        }
+        for (let i = 0; i < trades.length; i++) {
+            const trade = trades[i];
+            const parsed = this.parseTrade (trade, market);
```

Comment:
@pcriadoperez the `cost` seems off here, probably it's the `amount` here, can you double-check it pls?
![image](https://user-images.githubusercontent.com/43336371/195859779-fd274c9b-7efc-4bb3-90d8-a580863d030c.png)

Reply:
Great catch. Seems to be an existing issue with the `parseTrade` and `fetchTrades` function of deribit. It uses `safeTrade` to calculate the cost. However according to deribit the amount means something different depending on market type. `Trade amount. For perpetual and futures - in USD units, for options it is amount of corresponding cryptocurrency contracts, e.g., BTC or ETH.`

I think it makes sense to open a seperate PR for this

Reply:
@pcriadoperez so can you open it then?